nats/src/main.rs

```diff
@@ -310,3 +398,55 @@ impl Messaging for NatsMessagingProvider {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use wasmbus_rpc::provider::ProviderHandler;
+
+    /// Ensure that unlink triggers subscription removal
+    /// https://github.com/wasmCloud/capability-providers/issues/196
+    ///
+    /// NOTE: this is tested here for easy access to put_link/del_link without
+    /// the fuss of loading/managing individual actors in the lattice
+    #[tokio::test]
+    async fn test_unlink_unsub() {
+        // Build a nats messaging provider
+        let prov = NatsMessagingProvider::default();
+
+        // Actor should have no clients and no subs before hand
+        let actor_map = prov.actors.write().await;
```

Comment:
Nit (since this is a test), but you should only need a read lock here

Reply:
I wanted to force a write lock to ensure that no one else could be making changes -- i.e. this test would block until the work in `delete_link` is done.

Does that not make sense?

Reply:
Nope, that makes sense. Good idea